### ISSUE TITLE
PSBT_IN_WITNESS_UTXO incorrectly(?) exists for non-segwit inputs, confusing Coldcard

### ISSUE THREAD

**Issue by luke-jr** [spesmilo/electrum#8305]
Helping someone setup [Trezor/Ledger/Coldcard legacy P2SH multisig using Electrum](https://medium.com/@1700constantino/protect-your-bitcoin-with-an-electrum-multi-sig-wallet-with-coldcard-ledger-and-trezor-96397e4ce287), but when we get to the Coldcard signing step, it fails with "Failure Need witness script for input #0". Upon further investigation, we found that it is caused by PSBT_IN_WITNESS_UTXO being included for the non-segwit input(s), and manually removing that allowed the Coldcard to sign successfully.

@achow101 advised that it's undefined if this is valid or not. But since Coldcard apparently thinks it isn't, perhaps at least an export option that removes it would be nice (hex editing it out is a pain).

**Comment by SomberNight** [comment]
How are you exporting the psbt?

If it is via *not* via QR code, then this is unintended and will look into it more.

If it is via QR code, then it is an intentional hack.
https://github.com/spesmilo/electrum/blob/460c198b02d34c230c4f2572fc3f5674bde605ab/electrum/transaction.py#L941-L943

**Comment by luke-jr** [comment]
In this case, the transaction was created by Electrum, signed, sent via QR code to the second person's Electrum, then exported to a MicroSD for Coldcard to sign.

**Comment by luke-jr** [comment]
Also note *both* the non-witness UTXO and witness UTXO are in the PSBT

**Comment by SomberNight** [comment]
I think c9cc56b68785696fb32a727b3c94b1b274a4988c fixes it.
It's probably a regression from https://github.com/spesmilo/electrum/commit/d3227d7489fe327bd40e891a517c86bd207227ec.

Thanks for the report and explanation. I've added a test that simulates your scenario.

1. When exporting the QR code, the full utxo was converted to a witness utxo;
2. then when cosigner2 imported it, his wallet added the full utxo from the network, but kept the witness utxo also there;
3. and then on the sdcard the tx had both full utxo and witness utxo.

I've changed it so that in step2, when adding the full utxo, the witness utxo gets removed. (or in general, if we see a pre-segwit input that has both full utxo and witness utxo, we remove the witness utxo)
